### PR TITLE
Allow expungement date to be shifted to the future

### PR DESCRIPTION
**Pull request by KentShikama**
Sometimes we want to run our Expunger as if we were running it tomorrow or some arbitrary date in the future. We therefore add a new field internally called "today" to the SearchPanel to be able to change what day "today" is. By default it is the date as of now in PST/PDT. This is a really sweet feature addition from a technical perspective since we only rely on date.today() in one location in expunger.py, so we can do this change in relatively few lines and safely.

Here is a screenshot of the frontend changes:

Screen Shot 2021-07-03 at 1 02 31 PM

### CODE REVIEW

**Comment by michaelzhang43** [comment]
What an elegant solution!

**Review by michaelzhang43 — APPROVED** [review]
This sounds like a really elegant way to do it. Much better than the 'print all future eligible cases' idea I had. I especially like that this is the 'general solution' to my problem of being able to generate paperwork for one type of date (arbitrarily far in the future). Great idea

File: src/backend/expungeservice/endpoints/search.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+from datetime import datetime
+
```
Comment (wittejm) [inline src/backend/expungeservice/endpoints/search.py:2]: space.

File: src/backend/expungeservice/endpoints/search.py [hunk 2]
```diff
@@ -10,7 +12,7 @@
 from expungeservice.serializer import ExpungeModelEncoder
 from expungeservice.crypto import DataCipher
 from expungeservice.record_summarizer import RecordSummarizer
-from expungeservice.util import LRUCache
+from expungeservice.util import LRUCache, DateWithFuture as date_class
```
Comment (wittejm) [inline src/backend/expungeservice/endpoints/search.py:15]: I don't think this needs an alias.

**Review by wittejm — APPROVED** [review]
Two nitpicks, otherwise LGTM

File: src/backend/expungeservice/endpoints/search.py (hunk 2 above)
Comment (KentShikama) [reply src/backend/expungeservice/endpoints/search.py:15]: We're pretending like DateWithFuture is the standard `date` library, which we've called `date_class` across most of our codebase for legacy reasons. I'm not sure if I like this decision in retrospect but I'll stick with this for now until we feel like we should change it everywhere to not have an alias.

**Comment by michaelzhang43** [comment]
Bug: When I run it in dev, I can't generate paperwork if I change the expunge date. Had an intern try it and got the same result. The "Generate Paperwork" button still works, but does not generate any records

**Comment by KentShikama** [comment]
Ah yes, it is still generating paperwork as if "today" was now in PST. I've put a update here: https://github.com/codeforpdx/recordexpungPDX/pull/1529.

**Comment by NickSchimek** [comment]
This is sweet!

**Pull request merged**